peps/pep-0740.rst

```diff
@@ -269,25 +230,24 @@ following structure:
 
   All other keys in the ``publisher`` object are publisher-specific. A full
   illustrative example of a ``publisher`` object is provided in :ref:`appendix-2`.
-* ``attestations`` is a **required** JSON object, containing one or
-  more attestation objects as identified by their keys. This object is
-  a superset of ``attestations`` object supplied by the uploader through the
-  ``attestations`` field, as described in :ref:`upload-endpoint`.
+* ``attestations`` is a **required** JSON array, containing one or
+  more attestation objects. This array is a superset of ``attestations`` array
+  supplied by the uploader through the ``attestations`` field at upload time,
+  as described in :ref:`upload-endpoint`.
 
   Because ``attestations`` is a superset of the file's original uploaded attestations,
-  the index **MAY** chose to embed additional attestations of its own.
+  the index **MAY** chose to embed additional attestations. These attestations
+  may come from the index itself, from third parties, or as subsequent actions
+  produced by the original uploader.
 
 Simple Index
 ^^^^^^^^^^^^
 
 * When an uploaded file has one or more attestations, the index **MAY** include a
-  ``data-provenance`` attribute on its file link, with a value of either
-  ``true`` or ``false``.
-* When ``data-provenance`` is ``true``, the index **MUST** serve a
-  :ref:`provenance object <provenance-object>` at the same URL, but with
-  ``.provenance`` appended to it. For example, if ``HolyGrail-1.0.tar.gz``
-  exists and has associated attestations, those attestations would be located
-  within the provenance object hosted at ``HolyGrail-1.0.tar.gz.provenance``.
+  ``data-provenance`` attribute on its file link.
+
+* When the ``data-provenance`` attribute is present, its value **SHALL** be
+  a base64 encoded :ref:`provenance-object`.
```

Comment:
CC @di: this will embed the provenance object directly into the PEP 503 simple index as you suggested, but I want to call out that it'll likely be quite large there 🙂. I'll do some research to see how well/gracefully typical HTML5 parsers handle large data attributes.